### PR TITLE
Refresh: make pixel transitions a standalone component [fix #15524]

### LINKED ISSUE #15524: Refresh: Refactor transition component
Using an SVG mask for the pixel transition element has the drawback of requiring a solid fill color to overlap the content of the previous element. It also requires sufficient bottom padding on the previous element to allow for that overlap. In practice this has proven problematic. 

So instead of it being a mask on the top edge we think it'll be more practical and flexible for it to be its own component again, still with an SVG mask, but it can sit between elements and not try to overlap them. It'll mean we need some presentational classes for different color combinations but that's fine (it's a presentational anyway element).

### PR DESCRIPTION
**Pull request by craigcook**
## One-line summary
Changes the pixel transition from a decoration at the top of a section to a standalone element between sections. This solves problems of needing elements to overlap each other and needing to create space for that overlap. The trade-off is that this now needs to have background colors declared rather than inheriting from the adjacent sections, but that's a fair trade IMO.

## Issue / Bugzilla link
#15524 

## Testing
http://localhost:8000/en-US/
http://localhost:8000/en-US/about/

Ensure switch M24_WEBSITE_REFRESH is ON

### CODE REVIEW

**Comment by craigcook** [comment]
**For discussion:** For now I've added theme classes for pretty much the entire color palette. However, we probably don't need all these options at this stage and it could be reduced only to the backgrounds we've defined in https://github.com/mozilla/bedrock/blob/main/media/css/m24/theme.scss ... so do we go all the way with every color now, or only do the themes we need and we can always add more when/if we need more?

The more I think about it, I'm leaning toward the latter but wanted to pose the question before I made the change.

**Review by alexgibson — APPROVED** [review]
I like overall the approach here with the theme class names. I'm a fan of keeping things lean and only adding what we need, but there's not a lot of bloat here either way.

r+ (will leave to you to merge in case you wanted to remove any redundant theme classes)

**Comment by craigcook** [comment]
After sleeping on it, I did decide to rework the theme classes a bit to better align with the way we're applying themes elsewhere. I also renamed `m24-t-purple` to `m24-t-pink` just so the name matches the actual color (there is a different purple in the extended palette).

The changes are significant enough it probably warrants a re-review.

**Comment by craigcook** [comment]
Welp, this doesn't work in Safari. Marking Do Not Merge until I can sort that out.

**Comment by craigcook** [comment]
Alright, further refactoring to use a polygon directly in the CSS and avoid the fact that Safari doesn't support referencing an external SVG file.

File: bedrock/mozorg/templates/mozorg/home/includes/m24/media-springboard.html [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 {% from "macros-m24.html" import grid_tile, springboard_item with context %}
 
-<section class="m24-c-content m24-t-transition-01 m24-t-transition-next" id="media">
-  <h3 class="m24-c-intro-title m24-t-md">You, AI and the internet — what’s really going on?</h3>
+<section class="m24-c-content" id="media">
+  <h3 class="m24-c-intro-title m24-t-lg">You, AI and the internet — what’s really going on?</h3>
```
Comment (stephaniehobson) [inline bedrock/mozorg/templates/mozorg/home/includes/m24/media-springboard.html:10]:
```suggestion
  <h3 class="m24-c-intro-title m24-t-md">You, AI and the internet — what’s really going on?</h3>
```

.m24-t-md matches what's in the Figma.

**Review by stephaniehobson — APPROVED** [review]
R+wc

Please revert the change to the springboard title font size. Other than that this works great in everything I threw at it 🙂

**Comment by stephaniehobson** [comment]
Oooh, it let me merge my own suggestion.

**Pull request merged**